DOCS-211 Data transfer/migration tutorial draft (#59)
hazelcast/cloud-docs: docs/modules/ROOT/pages/shipping-logs-to-amazon-elasticsearch-service.adoc

= Sending Logs to Amazon OpenSearch
:description: You can set up a remote logging solution using Amazon OpenSearch Service. Logs from your Dedicated cluster are automatically sent to an Amazon OpenSearch domain. Amazon OpenSearch makes these logs available in Kibana.

:url-github-elasticsearch: https://github.com/elastic/elasticsearch
:url-amazon-opensearch: https://aws.amazon.com/opensearch-service
:url-aws-signup: https://portal.aws.amazon.com/billing/signup 
:url-aws-console: https://us-west-2.console.aws.amazon.com/es/home
:page-dedicated: true

{description}

link:{url-amazon-elasticsearch}[Amazon OpenSearch Service] is based on link:{url-github-elasticsearch}[Elasticsearch], and allows you to search, analyze and visualize your cluster logs using Kibana.

[[set-up-opensearch-domain]]
== Set Up an Amazon OpenSearch Domain

Create an Amazon OpenSearch domain to receive log files from your Hazelcast Cloud cluster.

. link:{url-aws-signup}[Create an AWS account] if you don't have one already.
. Sign in as root user.

> . From the list of regions on the top menu, select *US West (N California) us-west-1*.

Reviewer: The advice is to choose a region that's closest to your Hazelcast cluster to reduce latency.

Author: Updated as requested.

. In the *Network* section, select *Public access* to use fine-grained access control to secure the OpenSearch domain and data. VPC Peering connections between Hazelcast Cloud clusters and OpenSearch instances are not currently supported.
// Is this still true? 

. In the *Fine-grained access control* section, select *Create Master User* and enter a username and password for the account.

+
TIP: Make a note of the master user credentials. You'll need them later to access Kibana, and to make any configuration changes to your OpenSearch domain.
+

. In the *Access policy* section, select *Only use fine-grained access control*.
. Click *Create* to build your domain. You'll need to wait until your OpenSearch environment and the following URLs are created. You need both URLs to complete your logging setup.

** *Kibana URL*
** *Domain endpoint*

[[configure-cloud-logging]]
== Configure Logging Integration in Hazelcast Cloud

Configure your Hazelcast Dedicated cluster to send log files to the OpenSearch domain.

You'll need the master user credentials and domain endpoint that you created in xref:#configure-opensearch-domain[].

> . In Hazelcast Cloud, go to *Your Clusters* and select the cluster that want to integrate with OpenSearch.

Reviewer: ```suggestion
. Sign into the [.console]*link:{page-cloud-console}[{hazelcast-cloud} console]* and select the cluster that want to integrate with OpenSearch.
```

Reviewer: The `[.]` is called a role and it just allows us to style the sign-in button.

Author: Updated as requested.

. Select *Manage Cluster* > *Settings*, and on the *Cluster Settings* page select *Logging*.
. From the *Logging Technology* list, select *Elastic Stack (ELK)*.
. In the *Username* and *Password* fields, enter your master user credentials.
. In the *URL* field, enter the domain endpoint for your OpenSearch domain followed by port number 443. 
+ 
For example: `https://search-<domain-name>-lxxxxxxx3f2v3wz3tlkh4.us-west-1.es.amazonaws.com:443`
+
OpenSearch supports HTTPS over port 443.

. Click *Create Integration*. Within 30 seconds, Hazelcast starts to send log files to OpenSearch, which are made available in Kibana.

== Set Up a Kibana Dashboard

Create an index pattern to load the logs received from OpenSearch into a Kibana dashboard. 

> You'll need the master user credentials and Kibana URL that you created in xref:#configure-opensearch-domain[].

Reviewer: ```suggestion
You'll need the master user credentials and Kibana URL that you created when you <<configure-opensearch-domain, configured OpenSearch>>.
```

Author: Updated as requested.

[NOTE]
====

> To create log data for viewing, make sure that:

Reviewer: ```suggestion
To create log data for viewing, check for the following:
```

> * Your Hazelcast Cloud cluster is running 

Reviewer: ```suggestion
* Your {hazelcast-cloud} cluster is running 
```

* A client is writing data to the cluster.
====

> . From your domain details page in OpenSearch, click on the *Kibana URL*.

Reviewer: ```suggestion
. From your domain details page in OpenSearch, click *Kibana URL*.
```

> . Log in to Kibana using your master user credentials.

Author: Updated as requested.

> = Sending Logs to Amazon OpenSearch

Reviewer: ```suggestion
= Send Logs of Dedicated Clusters to Amazon OpenSearch
```

> :description: Set up a remote logging solution using Amazon OpenSearch Service. In this tutorial, you'll set up an Amazon OpenSearch domain and configure your {hazelcast-cloud} cluster to automatically send log files to it. You'll also learn how to view these logs in Kibana.

Author: Updated

. Select *Explore on my own* to skip onboarding.
. From the *Select tenant* dialog box, select *Private* and click *Confirm*.
. From the main menu, select *Kibana* > *Discover* to start your set up.
. Click *Create index pattern*. An index pattern tells Kibana which indices contain the data that you want to view.
. In the *Index pattern* field, enter a wildcard `*` to find all log data received from OpenSearch. 
. When Kibana finds the {hazelcast-cloud} logs, click *Next step*.
. From the *Time field* list, select *updated_at* so that you'll be able to filter your log file data by time.
. Click *Create index pattern*. Your log file data is ready to view. Simply select *Kibana* > *Discover* from the main menu.

+
image:kibana-dashboard.png[Log data in Kibana Dashboard]

If necessary, you can now return to your OpenServe domain and increase the number of nodes or availability zones to add resilience to the logging service.

== Summary

In this tutorial, you learned how to do the following:

* Create an Amazon OpenSearch domain to receive log files from your {hazelcast-cloud} cluster.
* Configure your Dedicated cluster to automatically send log files to the OpenSearch domain.

> * Load log files into a Kibana dashboard using an index pattern.

Reviewer: Would be good to include next steps or related resources so that users can go somewhere else after the tutorial.

= Send Logs of Dedicated Clusters to Amazon OpenSearch
:description: Set up a remote logging solution using Amazon OpenSearch Service. In this tutorial, you'll set up an Amazon OpenSearch domain and configure your {hazelcast-cloud} cluster to automatically send log files to it. You'll also learn how to view these logs in Kibana.
:page-dedicated: true
:url-github-elasticsearch: https://github.com/elastic/elasticsearch
:url-amazon-opensearch: https://aws.amazon.com/opensearch-service
:url-aws-signup: https://portal.aws.amazon.com/billing/signup 
:url-aws-console: https://us-west-2.console.aws.amazon.com/es/home

{description}

link:{url-amazon-elasticsearch}[Amazon OpenSearch Service] is based on link:{url-github-elasticsearch}[Elasticsearch], and allows you to search, analyze and visualize your cluster logs using Kibana.

[[set-up-opensearch-domain]]
== Step 1. Set Up an Amazon OpenSearch Domain

Create an Amazon OpenSearch domain to receive log files from your {hazelcast-cloud} cluster.

. link:{url-aws-signup}[Create an AWS account] if you don't have one already.

> . Sign in as root user.

Reviewer 2: Maybe it might seem trivial, but I don't think this is a good idea to encourage user to login via their root users. Instead, we can tell them to create IAM user and login with IAM credentials. You can see the details from AWS here https://docs.aws.amazon.com/IAM/latest/UserGuide/id_root-user.html.

Author: Could we suggest that they use root user just for testing purposes but then create an IAM account for Production?

The reason for this, is that I started to look into the account configuration and it’s not that straightforward. It looks like you need to create an IAM account with access to the OpenSearchService through a policy, group or similar and somehow enable a service-linked role for it. I couldn’t work out how to enable the service-linked role.

:page-dedicated: true
:url-github-elasticsearch: https://github.com/elastic/elasticsearch
:url-amazon-opensearch: https://aws.amazon.com/opensearch-service
:url-aws-signup: https://portal.aws.amazon.com/billing/signup 
:url-aws-console: https://us-west-2.console.aws.amazon.com/es/home
:url-aws-managed-policy: https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ac-managed.html 
:url-aws-iam-user: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html

{description}

link:{url-amazon-elasticsearch}[Amazon OpenSearch Service] is based on link:{url-github-elasticsearch}[Elasticsearch], and allows you to search, analyze and visualize your cluster logs using Kibana.

[[set-up-opensearch-domain]]
== Step 1. Set Up an Amazon OpenSearch Domain

Create an Amazon OpenSearch domain to receive log files from your {hazelcast-cloud} cluster.

. link:{url-aws-signup}[Create an AWS account] if you don't have one already.

> . For testing purposes, you can sign in as root user but we recommend that you sign in with an link:{url-aws-iam-user}[IAM account] for Production use.

Reviewer 2: `... user, but we recommend...`
